src/Catalyst.Core.Modules.Ledger.Tests/UnitTests/LedgerTests.cs

```diff
@@ -98,16 +122,63 @@ public void Should_Reconcile_On_New_Delta_Hash()
+            _ledgerSynchroniser.CacheDeltasBetween(Arg.Is(_genesisHash), Arg.Is(hash1), default)
                .ReturnsForAnyArgs(new[] { hash2, hash1, _genesisHash });
 
             _deltaHashProvider.DeltaHashUpdates.Returns(updates.ToObservable(_testScheduler));
 
-            _ledger = new LedgerService(_executor, _stateProvider, _storageProvider, new StateDb(), new StateDb(), _fakeRepository, _deltaHashProvider, _ledgerSynchroniser, _mempool, _logger);
+            _ledger = new LedgerService(_executor, _stateProvider, _storageProvider, new StateDb(), new StateDb(), _fakeRepository, _deltaHashProvider, _ledgerSynchroniser, _mempool, _mapperProvider, _logger);
+
+            _testScheduler.Start();
+
+            _ledger.LatestKnownDelta.Should().Be(_genesisHash);
+        }
+
+        private IEnumerable<PublicEntry> GenerateSamplePublicTransactions(int sampleSize)
+        {
+            for (var i = 0; i < sampleSize; i++)
+            {
+                var sender = _cryptoContext.GeneratePrivateKey();
+                var recipient = _cryptoContext.GeneratePrivateKey().GetPublicKey();
+                var publicEntry = EntryUtils.PreparePublicEntry(recipient, sender.GetPublicKey(), 10);
+                publicEntry.Signature = publicEntry.GenerateSignature(_cryptoContext, sender, _signingContext);
+                yield return publicEntry;
+            }
+        }
+
+        [Fact]
+        public void Should_Delete_MempoolItems_On_New_Delta_Hash()
+        {
+            var sampleSize = 5;
+            _mempool.Service.Returns(new MempoolService(new InMemoryRepository<PublicEntryDao, string>()));
+
+            var hash = CidHelper.CreateCid(_hashProvider.ComputeUtf8MultiHash("update"));
+            var updates = new[] { hash };
+
+            _ledgerSynchroniser.CacheDeltasBetween(Arg.Is(_genesisHash), Arg.Is(hash), default)
+               .ReturnsForAnyArgs(new[] { hash, _genesisHash });
+
+            var allPublicEntries = GenerateSamplePublicTransactions(sampleSize * 2).ToList();
+
+            //Add all public entries to the mempool
+            allPublicEntries.Select(x => x.ToDao<PublicEntry, PublicEntryDao>(_mapperProvider)).ToList().ForEach(x => _mempool.Service.CreateItem(x));
+
+            //Only add half of all public entries to the delta
+            var delta = new Delta()
+            {
+                TimeStamp = Timestamp.FromDateTime(DateTime.UtcNow),
+                PublicEntries = { allPublicEntries.Take(sampleSize) }
+            };
+
+            _ledgerSynchroniser.DeltaCache.TryGetOrAddConfirmedDelta(Arg.Is(hash), out Arg.Any<Delta>()).Returns(x => { x[1] = delta; return true; });
+
+            _deltaHashProvider.DeltaHashUpdates.Returns(updates.ToObservable(_testScheduler));
+
+            _ledger = new LedgerService(_executor, _stateProvider, _storageProvider, new StateDb(), new StateDb(), _fakeRepository, _deltaHashProvider, _ledgerSynchroniser, _mempool, _mapperProvider, _logger);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Split this 209 characters long line (which is greater than 200 authorized).](https://app.codacy.com/app/catalyst-network/Catalyst.Framework/pullRequest?prid=4562326)